src/grt/src/GlobalRouter.cpp

```diff
@@ -221,17 +221,27 @@ void GlobalRouter::saveCongestion()
   }
   std::ofstream out(congestion_file_name_);
 
-  std::vector<std::pair<GSegment, TileCongestion>> congestionGridsV,
-      congestionGridsH;
+  std::vector<std::tuple<GSegment, TileCongestion, std::vector<std::string>>>
+      congestionGridsV, congestionGridsH;
   fastroute_->getCongestionGrid(congestionGridsV, congestionGridsH);
   for (auto& it : congestionGridsH) {
+    GSegment seg;
+    TileCongestion tile;
+    std::vector<std::string> srcs;
+    std::tie(seg, tile, srcs) = it;
     out << "violation type: Horizontal congestion\n";
-    const int capacity = it.second.first;
-    const int usage = it.second.second;
-    out << "\tsrcs: \n";
+    const int capacity = tile.first;
+    const int usage = tile.second;
+    out << "\tsrcs: ";
+    if (!srcs.empty()) {
+      for (const auto& net : srcs) {
+        out << "net:" << net << " ";
+      }
+    }
```

Comment:
Why check if it is empty as the loop won't do anything in that case.  Does it make sense to print srcs: if there are none?  I'm not certain what the parser expects.

Reply:
I've moved the check above and will not print `"srcs: "` if it is empty.